### PR TITLE
Bumped Mesos and added patches to extra/patches.

### PR DESCRIPTION
**Pull request by karya0**
## High-level description

There are two changes:
1. A routine Mesos bump based on https://github.com/dcos/dcos/pull/2808.
2. Mesos buildinfo.json now points to apache/mesos git repo/sha. The cherry-pick commits have been added to mesos/extra/patches directory.

## Related tickets (optional)

Other tickets related to this change:

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [Mesos changelog](https://github.com/apache/mesos/compare/e91ce42ed56c5ab65220fbba740a8a50c7f835ae...a1c6a7a3c54518f97a34f28b1792885b928b948c)
  - [ ] Test Results:
  - [ ] Code Coverage (if available):
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by karya0** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by karya0** [comment]
@mesosphere-mergebot label Work in Progress

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot2** [comment]
@karya0, you must be a member of the Mesosphere organization to issue this command.

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by karya0** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot2** [comment]
Your pull request's branch is not based on the most recent version of `master`. Please rebase your changes against this repo's `master` branch.

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot-dev** [comment]
:wave: This PR has been inactive for 3 days. Please review the status checks to see what needs to be done to move it forward or change the label to Work in Progress. Thank you!

File: packages/mesos/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <h2>Patches to cherry-pick on top of open source Apache Mesos before building in DC/OS</h2>
 
-See [patches.json](patches.json) for the list of patches.
+See [extra/patches](patches) directory for the list of patches.
```
Comment (rukletsov) [inline packages/mesos/README.md]: I think we should simply kill this file. "extra/patches" is self-explanatory.

File: packages/mesos/build [hunk 2]
```diff
@@ -3,6 +3,12 @@ libdir="$PKG_PATH/lib"
 
 # TODO(cmaloney): Check prerequisites installed (glog, protobuf, boost)
 pushd "/pkg/src/mesos"
+
+# Apply patches from packages/mesos/patch directory. All patches are numbered
+for patch in /pkg/extra/patches/*; do
+  git -c user.name="Mesosphere CI" -c user.email="mesosphere-ci@users.noreply.github.com" am $patch
```
Comment (rukletsov) [inline packages/mesos/build]: Would "Build Robot" be a better name? Or "Mesosphere CI" is what we use?

File: packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+From 0f762c15455900083f7a3f72c6cdc60a418ec12f Mon Sep 17 00:00:00 2001
```
Comment (rukletsov) [inline packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch]: Does it make sense / can we remove this line? It seems to me that there is no value in this info.

File: packages/mesos/README.md (hunk 1 above)
Comment (karya0) [reply packages/mesos/README.md]: I agree that `extra/patches` is self-explanatory but it's one level too deep and not visible to someone looking at this directory for Mesos sources.

File: packages/mesos/build (hunk 2 above)
Comment (karya0) [reply packages/mesos/build]: That's the "name" associated with the github user so I just used that.

File: packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch (hunk 3 above)
Comment (karya0) [reply packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch]: It is what git-format reported. I am not sure what's the convention there.

File: packages/mesos/README.md (hunk 1 above)
Comment (gpaul) [reply packages/mesos/README.md]: I think he meant
```
See [extra/patches](extra/patches) directory for the list of patches.
```
/me sneaks back into the shadows...

Comment (rukletsov) [reply packages/mesos/README.md]: I meant killing the file altogether. However now I think it makes sense to tell folks how to prepare new patches, e.g., use `git format-patch --stdout <range> | sed 1d` to omit the first line.

File: packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch (hunk 3 above)
Comment (bbannier) [reply packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch]: Could we please just use whatever `git-format` reports? This e.g., allows to construct proper history with `git-am` (which would be easier to correlate with upstream).

`patch` will be fine with a surprisingly large collection of formats.

File: packages/mesos/README.md (hunk 1 above)
Comment (karya0) [reply packages/mesos/README.md]: I think something like the following should suffice:

<h2>Mesos Patches</h2>

We need to apply some patches on top of upstream Apache Mesos before building in
DC/OS.  The patches are stored in the [extra/patches] directory and applied as part of [build] script.

Here are the instructions to add/remove/update one or more patches:
* Checkout the upstream Mesos SHA.
* Apply all the cherry-picks that you need.
* Create patch files using `git format-patch -N <upstream-SHA>`
* Replace existing files in [extra/patches] with the new patche files.

File: packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch (hunk 3 above)
Comment (karya0) [reply packages/mesos/extra/patches/0001-Set-LIBPROCESS_IP-into-docker-container.patch]: I'd prefer to stick with `git-format` output -- fewer troubles that way.

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot-dev** [comment]
:wave: This PR has been inactive for 3 days. Please review the status checks to see what needs to be done to move it forward or change the label to Work in Progress. Thank you!

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot-dev** [comment]
Your pull request's branch is not based on the most recent version of `master`. Please rebase your changes against this repo's `master` branch.

**Comment by karya0** [comment]
@mesosphere-mergebot bump-ee

**Pull request merged**